lib/dm-core/property/object.rb

```diff
@@ -11,6 +11,8 @@ def dump(value)
 
       # @api semipublic
       def load(value)
+        return if value.nil?
```

Comment:
I believe we want to revert the changes to `typecast` that occurred in `lib/dm-core/property.rb` instead of the change here.
I.e. change that back to
```
    def typecast(value)
      if (value.nil? || value_loaded?(value))
        value
      elsif respond_to?(:typecast_to_primitive, true)
        typecast_to_primitive(value)
      else
        value
      end
    end
```